include/boost/geometry/algorithms/detail/equals/implementation.hpp

```diff
@@ -226,50 +246,42 @@ struct equals_by_relate
         >
 {};
 
-// Small helper structure do decide to use collect_vectors, or not
-template <typename Strategy, typename CsTag>
-struct use_collect_vectors
-{
-    static constexpr bool value = false;
-};
-
-template <typename Strategy>
-struct use_collect_vectors<Strategy, cartesian_tag>
-{
-    static constexpr bool value = true;
-};
-
-template <typename CV, typename CsTag>
-struct use_collect_vectors<strategy::side::spherical_side_formula<CV>, CsTag>
-{
-    static constexpr bool value = true;
-};
-
 // Use either collect_vectors or relate
 // NOTE: the result could be conceptually different for invalid
 // geometries in different coordinate systems because collect_vectors
 // and relate treat invalid geometries differently.
 template<typename TrivialCheck>
 struct equals_by_collection_or_relate
 {
-    template <typename Geometry1, typename Geometry2, typename Strategy>
+    template <typename Strategy>
+    using use_vectors = use_collect_vectors
+        <
+            decltype(std::declval<Strategy>().side()),
+            typename Strategy::cs_tag
+        >;
+
+    template
+    <
+        typename Geometry1, typename Geometry2, typename Strategy,
+        std::enable_if_t<use_vectors<Strategy>::value, int> = 0
+    >
     static inline bool apply(Geometry1 const& geometry1,
                              Geometry2 const& geometry2,
                              Strategy const& strategy)
     {
-        using side_strategy = decltype(std::declval<Strategy>().side());
-        using implementation = std::conditional_t
-            <
-                use_collect_vectors
-                <
-                    side_strategy,
-                    typename Strategy::cs_tag
-                >::value,
-                equals_by_collection<TrivialCheck>,
-                equals_by_relate<Geometry1, Geometry2>
-            >;
-
-        return implementation::apply(geometry1, geometry2, strategy);
+        return equals_by_collection<TrivialCheck>::apply(geometry1, geometry2, strategy);
+    }
+
+    template
+    <
+        typename Geometry1, typename Geometry2, typename Strategy,
+        std::enable_if_t<! use_vectors<Strategy>::value, int> = 0
+    >
+    static inline bool apply(Geometry1 const& geometry1,
+                             Geometry2 const& geometry2,
+                             Strategy const& strategy)
+    {
+        return equals_by_relate<Geometry1, Geometry2>::apply(geometry1, geometry2, strategy);
     }
```

Comment:
Thanks for this! I was already thinking how the two structures could be combined, but didn’t get this yet. Looks good!